numba/_helperlib.c

```diff
@@ -977,18 +847,122 @@ numba_do_raise(PyObject *exc_packed)
         traceback_add(function_name_str, filename_str, \
                       (int)PyLong_AsLong(lineno));
     }
+}
 
-    /* PyErr_SetObject incref's its arguments */
-    Py_XDECREF(value);
-    Py_XDECREF(type);
-    return 0;
+static
+int reraise_exc_is_none() {
```

Comment:
Would be good to expand the comment to include information about what this function does.

Reply:
Done in https://github.com/numba/numba/pull/6017/commits/07415ebafbe9ae43e20e824762ce6e3941b4f21c